github/repos_test.go

```diff
@@ -4463,3 +4464,49 @@ func TestRepositoriesService_IsPrivateReportingEnabled(t *testing.T) {
 		return resp, err
 	})
 }
+
+func TestRepository_UnmarshalJSON(t *testing.T) {
+	var testCases = map[string]struct {
+		data           []byte
+		wantRepository Repository
+		wantErr        bool
+	}{
+		"Empty": {
+			data:           []byte("{}"),
+			wantRepository: Repository{},
+			wantErr:        false,
+		},
+		"Invalid JSON": {
+			data:           []byte("{"),
+			wantRepository: Repository{},
+			wantErr:        true,
+		},
+		"Partial project": {
+			data:           []byte(`{"id":10270722,"name":"go-github","private":false,"owner":{"login":"google"},"created_at":"2013-05-24T16:42:58Z","license":{},"topics":["github"],"permissions":{"pull":true},"custom_properties":{},"organization":{"login":"google"}}`),
+			wantRepository: Repository{ID: Int64(10270722), Name: String("go-github"), Private: Bool(false), Owner: &User{Login: String("google")}, CreatedAt: &Timestamp{time.Date(2013, 5, 24, 16, 42, 58, 0, time.UTC)}, License: &License{}, Topics: []string{"github"}, Permissions: map[string]bool{"pull": true}, CustomProperties: map[string]interface{}{}, Organization: &Organization{Login: String("google")}},
+			wantErr:        false,
+		},
+		"With custom properties": {
+			data:           []byte(`{"custom_properties":{"boolean":"false","text":"a","single-select":"a","multi-select":["a","b","c"]}}`),
```

Comment:
How about this?

```suggestion
			data:           []byte(`{"custom_properties":{"boolean":false,"int":42,"key":"value","array":["a","b","c"]}}`),
```

Reply:
This would imply that there is a possible integer value when there is non (that we know of). I used the different custom property types offered by GitHub (see [here](https://docs.github.com/en/organizations/managing-organization-settings/managing-custom-properties-for-repositories-in-your-organization#:~:text=This%20can%20either%20be%20a%20text%20string%2C%20a%20single%20select%20field%2C%20a%20multi%20select%20field%2C%20or%20a%20true/false%20boolean)) and observed their representation in the api, using their label in the github ui as key for this json.

I guess it depends on whats the goal - testing that the unmarshalling works according to the type or testing the possible values that github could send for the given field. What do you think?

Reply:
Oh, I see, well at least please change the boolean from a string to `false` without the quotes.

Reply:
GitHub returns the value for boolean type custom properties as strings ("true" or "false"). Should I remove the quotes regardless?

Reply:
Are you sure?  We have hundreds of endpoints that return booleans, and none of them are strings.
What makes you say this?  You made a boolean custom property and it is returning a string?!?

Reply:
Yes, I made custom properties of all types and looked at the api response from github.
This is the actual response from the github api with one custom property (named accordingly) of each type configured:

```jsonc
{
  // ...
   "custom_properties": {
      "type-multi-select": [
        "a",
        "b"
      ],
      "type-single-select": "b",
      "type-text": "lala",
      "type-true-false": "true"
  },
  //...
}
```

Reply:
I'd love if they would return an actual boolean here :smile:

Reply:
OK, bizarre.
Then it looks like there is one linting failure remaining... let me look.

Reply:
It looks like the generated files are out-of-date if you could please update them.

Reply:
The whole integration of file generation is a really cool golang feature!
Just pushed the updated files :+1: